dmriprep/interfaces/vectors.py

```diff
@@ -1,91 +1,101 @@
-            fbval = self.inputs.fbval
-            fbvec = self.inputs.fbvec
-        else:
-            fbval = None
-            fbvec = None
-
-        if isdefined(self.inputs.rasb_tsv_file):
-            rasb_tsv_file = self.inputs.rasb_tsv_file
-        else:
-            rasb_tsv_file = None
-
-        if isdefined(self.inputs.basedir):
-            basedir = self.inputs.basedir
-        else:
-            basedir = runtime.cwd
-
-        vt = VectorTools(basedir, self.inputs.dwi_file, fbval, fbvec, rasb_tsv_file, self.inputs.B0_NORM_EPSILON,
-                         self.inputs.bval_scaling)
-
-        # Read in vectors
-        if rasb_tsv_file is not None:
-            vt.read_rasb_tsv()
-        elif fbval is not None and fbvec is not None:
-            vt.read_bvals_bvecs()
-        else:
-            raise OSError('VectorTools methods require either fbval/fbvec input files or an input rasb .tsv file.')
-
-        # Rescale vectors
-        if self.inputs.rescale is True:
-            vt.rescale_vecs()
-
-        # Build gradient table
-        gtab, b0_ixs = vt.build_gradient_table()
-
-        # Check vector integrity
-        is_hemi, pole, slm = vt.checkvecs()
-
-        if self.inputs.save_fsl_style is True:
-            fbval_out_file, fbvec_out_file = vt.save_vecs_fsl()
-        else:
-            fbval_out_file = None
-            fbvec_out_file = None
-
-        self._results['gtab'] = gtab
-        self._results['is_hemi'] = is_hemi
-        self._results['pole'] = pole
-        self._results['slm'] = slm
-        self._results['b0_ixs'] = b0_ixs
-        self._results['fbval_out_file'] = fbval_out_file
-        self._results['fbvec_out_file'] = fbvec_out_file
-        self._results['rasb_tsv_out_file'] = vt.write_rasb_tsv()
+        rasb_file = _undefined(self.inputs, 'in_rasb')
+
+        table = DiffusionGradientTable(
+            self.inputs.dwi_file,
+            bvecs=_undefined(self.inputs, 'in_bvec'),
+            bvals=_undefined(self.inputs, 'in_bval'),
+            rasb_file=rasb_file,
+            b_scale=self.inputs.b_scale,
+            bvec_norm_epsilon=self.inputs.bvec_norm_epsilon,
+            b0_threshold=self.inputs.b0_threshold,
+        )
+        pole = table.pole
+        self._results['pole'] = tuple(pole)
+        self._results['full_sphere'] = np.all(pole == 0.0)
+        self._results['b0_ixs'] = np.where(table.b0mask)[0].tolist()
+
+        if rasb_file is None:
+            rasb_file = fname_presuffix(
+                self.inputs.dwi_file, use_ext=False, suffix='.tsv',
+                newpath=runtime.cwd)
+            table.to_filename(rasb_file)
+        self._results['out_rasb'] = rasb_file
+
+        cwd = Path(runtime.cwd).absolute()
+
+        table.to_filename(bvecs=cwd / 'bvec', bvals=cwd / 'bval')
+        self._results['out_bval'] = str(cwd / 'bval')
+        self._results['out_bvec'] = str(cwd / 'bvec')
         return runtime
+
+
+def _undefined(objekt, name, default=None):
```

Comment:
this should be part of nipype probably :P

Reply:
That would be nice!